### DISCUSSION TITLE
Can I generate documentation from the validator?

### DISCUSSION THREAD

**Discussion by krankenberg** [making/yavi#272] [Q&A] [answered]
Hello,

currently, I am building a rather complex validator with this library and I am thinking about how I will document those validations, so users of the API can see what values are needed for what conditions etc.

Is there any possibility to generate some sort of documentation from a validator which will list all constraints?

**Comment by making** [comment, marked as answer]
At this time, YAVI does not have such a feature. I think it is technically possible.
Actually, investigating whether it can work with OpenAPI is one of the Todo lists in my head. I haven't got enough time so I haven't started yet ...

**Reply by krankenberg** [reply]
Yes, documentation via OpenAPI would be awesome.
Thanks for answering, already figured that.
